lib/metalsmith.js

```diff
@@ -20,12 +21,45 @@ const views = [
   paths.govukfrontend
 ]
 
+const coloursAppliedSCSS = scssToJson('./node_modules/@govuk-frontend/globals/settings/_colours-applied.scss', {
+  dependencies: [{path: './node_modules/@govuk-frontend/globals/settings/_colours-palette.scss'}]
+})
+
+const coloursPaletteSCSS = scssToJson('./node_modules/@govuk-frontend/globals/settings/_colours-palette.scss')
+
+const coloursSCSS = Object.assign(coloursAppliedSCSS, coloursPaletteSCSS)
+
+console.dir(coloursSCSS)
+
+// process colours data
+
+var colours = require('../data/colours.json')
+
+for (const groupName in colours){
+  var group = colours[groupName]
+  if (Array.isArray(group)){
```

Comment:
Given the two arrays need to be handled differently, wonder if it's worth not trying to merge them into one array at all – and just having two different variables for the palette and applied colours that get passed into the view. We basically have the two different for loops to handle each one anyway.